source/dub/generators/generator.d

```diff
@@ -1051,3 +1053,51 @@ private void storeRecursiveInvokations(ref const(string[string])[] env, string[]
 			.join(",")
 	];
 }
+
+version(Posix) {
+    // https://github.com/dlang/dub/issues/2238
+	unittest {
+		import dub.internal.vibecompat.data.json : parseJsonString;
+		import dub.compilers.gdc : GDCCompiler;
+		import std.algorithm : canFind;
+		import std.path : absolutePath;
+		import std.file : mkdirRecurse, rmdirRecurse, write;
+
+		mkdirRecurse("dubtest/preGen/source");
```

Comment:
I don't think the dub unittests write to the filesystem. It would probably be better to use the tests folder test system with shell scripts.

Reply:
Is that a good enough reason? I think it's a lot better to unit test even if it requires writing to the file system, especially as the only way I can get this to manifest in real life is with two dub packages where one depends on the other.

Reply:
my main reason would be that all the existing similar tests use the test folder (shell scripts or D scripts, optionally with their own folders where they can do whatever they want) and I don't think dub has such tests inside the unittests.

Additionally the test runner in the tests folder is more easily usable with isolating tests and debugging issues.

Example test script:
https://github.com/dlang/dub/blob/059f7a2a16cf7a50a5c68dc295c796482116ff87/test/environment-variables.script.d

and test folder:
https://github.com/dlang/dub/tree/059f7a2a16cf7a50a5c68dc295c796482116ff87/test/environment-variables

Reply:
Can we please merge? I don't think "We've always done it this way" is a valid argument.